src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -107,9 +107,20 @@ public async Task ConnectAsync(Uri uri, HttpMessageInvoker? invoker, Cancellatio
                             externalAndAbortCancellation = _abortSource;
                         }
 
+                        Task<HttpResponseMessage> sendTask;
+                        if (invoker is HttpClient)
+                        {
+                            HttpClient? client = invoker as HttpClient;
+                            sendTask = client!.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, externalAndAbortCancellation.Token);
```

Comment:
Maybe just use `is` without `as`:

```suggestion
                        if (invoker is HttpClient client)
                        {
                            sendTask = client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, externalAndAbortCancellation.Token);
```